presto-spi/src/main/java/com/facebook/presto/spi/statistics/RangeColumnStatistics.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.statistics;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.spi.statistics.Estimate.unknownValue;
+import static java.util.Collections.unmodifiableMap;
+import static java.util.Objects.requireNonNull;
+
+public class RangeColumnStatistics
+{
+    public static final String DATA_SIZE_STATISTIC_KEY = "data_size";
+    public static final String FRACTION_STATISTICS_KEY = "fraction";
+    public static final String DISTINCT_VALUES_STATISTICS_KEY = "distinct_values_count";
+
+    private final Optional<Object> lowValue;
+    private final Optional<Object> highValue;
+    private final Map<String, Estimate> statistics;
+
+    public RangeColumnStatistics(
+            Optional<Object> lowValue,
+            Optional<Object> highValue,
+            Estimate fraction,
+            Estimate dataSize,
+            Estimate distinctValuesCount)
+    {
+        this.lowValue = requireNonNull(lowValue, "lowValue can not be null");
+        this.highValue = requireNonNull(highValue, "highValue can not be null");
+        requireNonNull(fraction, "fraction can not be null");
+        requireNonNull(dataSize, "dataSize can not be null");
+        requireNonNull(distinctValuesCount, "distinctValuesCount can not be null");
```

Comment:
you can defer the checks to "assignment" in `createStatisticsMap`

(unless you follow @kokosing advice (which i agree with) not to have a map here)